shop3/looping-tasks/exts-common.lisp

```diff
@@ -0,0 +1,89 @@
+(in-package :shop)
+
+(defclass looping-domain (domain)
```

Comment:
This isn't a mixin as defined.  Suggest rewrite:

```
(defclass looping-mixin ()
  ())

(defclass looping-domain (looping-mixin domain)
  ())
```

Then you need to rewrite your methods to dispatch on `looping-mixin`, not `looping-domain`.